src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -204,30 +192,40 @@
                        "listed in PuppetDB's certificate-whitelist file?")
                   ssl-client-cn))))))
 
-(defn shutdown-mq-broker
+(defn shutdown-mq
   "Explicitly shut down the queue `broker`"
-  [{:keys [broker]}]
+  [{:keys [broker mq-factory mq-connection]}]
   (when broker
-    (log/info "Shutting down message broker.")
-    ;; Stop the mq the old-fashioned way
+    (log/info "Shutting down the messsage queues.")
+    (.close mq-connection)
+    (.close mq-factory)
     (mq/stop-broker! broker)))
 
+(defn shutdown-updater
+  [context]
+  (when-let [updater (context :updater)]
+    (log/info "Shutting down updater thread.")
+    (future-cancel updater)
+    (let [timeout (Object.)]
+      (try
+        (when (= timeout (deref updater 3000 timeout))
+          (log/error "waited for updater thread for 3s but it didn't finish"))
```

Comment:
@ajroetker I believe this will only log an error if the updater thread can't be cancelled.  Though now that you mention it, perhaps it needs a clause like this too:
```
        (catch java.io.IOError ex ;; Covers situation where there's no net.
          true)
```